packages/ampd-sdk/src/grpc/utils.rs

```diff
@@ -0,0 +1,114 @@
+use ampd_proto;
+use ampd_proto::{BroadcastResponse, ContractsResponse, KeyId};
+use axelar_wasm_std::nonempty;
+use cosmrs::AccountId;
+use error_stack::{Report, Result, ResultExt as _};
+use report::ResultCompatExt;
+use tokio::sync::{mpsc, watch};
+use tonic::transport;
+
+use crate::grpc::error::{AppError, Error};
+
+#[derive(Debug, Clone)]
+pub enum ClientMessage {
+    ConnectionFailed(String),
+}
+
+#[derive(Debug, Clone)]
+pub enum ConnectionState {
+    Connected(transport::Channel),
+    Disconnected,
+    Reconnecting,
+}
+
+impl PartialEq for ConnectionState {
+    fn eq(&self, other: &Self) -> bool {
+        matches!(
+            (self, other),
+            (ConnectionState::Connected(_), ConnectionState::Connected(_))
+                | (ConnectionState::Disconnected, ConnectionState::Disconnected)
+                | (ConnectionState::Reconnecting, ConnectionState::Reconnecting)
+        )
+    }
```

Comment:
use std::mem::discriminant for comparison instead, then you don't have to write a brittle check that will become wrong if we add different variants